### PR TITLE
Create save_gdal.py

### PR DESCRIPTION
**Pull request by 4restwilliams**
The creation of save_geotiff.py allows for the export of mintpy files to geotiff files for analysis in external gis programs.

### CODE REVIEW

**Comment by yunjunz** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mintpy/save_geotiff.py  3
         

Clones added
============
- mintpy/save_geotiff.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/insarlab/MintPy/pullRequest?prid=5643553&bid=18477775)

File: mintpy/save_geotiff.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+def array2raster(newRasterfn,rasterOrigin,pixelWidth,pixelHeight,array):
+
+    cols = array.shape[1]
+    rows = array.shape[0]
+    originX = rasterOrigin[0]
+    originY = rasterOrigin[1]
+
+    driver = gdal.GetDriverByName('GTiff')
+    outRaster = driver.Create(newRasterfn, cols, rows, 1, gdal.GDT_Float64)
+    outRaster.SetGeoTransform((originX, pixelWidth, 0, originY, 0, pixelHeight))
```
[the remark this answers is not in the document]
Comment (hfattahi) [reply mintpy/save_geotiff.py:52]: Awesome! so no need to change this part.

File: mintpy/save_geotiff.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    return inps
+
+def array2raster(newRasterfn,rasterOrigin,pixelWidth,pixelHeight,array):
+
+    cols = array.shape[1]
+    rows = array.shape[0]
+    originX = rasterOrigin[0]
+    originY = rasterOrigin[1]
+
+    driver = gdal.GetDriverByName('GTiff')
```
[the remark this answers is not in the document]
Comment (4restwilliams) [reply mintpy/save_geotiff.py:50]: Hi @hfattahi, yes I'll make these edits

File: mintpy/save_geotiff.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+def main(iargs=None):
+    inps = cmd_line_parse(iargs)
+
+    if not inps.dset:
+        array, attr = readfile.read(inps.file)
+    array, attr = readfile.read(inps.file, datasetName=inps.dset)
+
+    rasterOrigin = (float(attr['X_FIRST']),float(attr['Y_FIRST']))
+    pixelWidth = float(attr['X_STEP'])
+    pixelHeight = float(attr['Y_STEP'])
```
[the remark this answers is not in the document]
Comment (4restwilliams) [reply mintpy/save_geotiff.py:69]: Hi @hfattahi, yes I'll make these edits

**Comment by yunjunz** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mintpy/save_gdal.py  3
         

Clones added
============
- mintpy/save_gdal.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/insarlab/MintPy/pullRequest?prid=5643553&bid=18477775)

**Comment by yunjunz** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mintpy/save_gdal.py  7
         

Clones added
============
- mintpy/save_gdal.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/insarlab/MintPy/pullRequest?prid=5643553&bid=18477775)

**Comment by 4restwilliams** [comment]
Happy to contribute!

**Pull request merged**